### PR TITLE
Initial code

### PR DESCRIPTION
**Pull request by sreynen**
This is the full project, so I won't repeat the README and other documentation here.

### CODE REVIEW

File: post_report.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    if len(approved_accounts):
+        args.SLACK_MESSAGE_TEXT = "The following pending Mobilize accounts "
+        args.SLACK_MESSAGE_TEXT += "should be *approved* "
+        args.SLACK_MESSAGE_TEXT += "(subscribed to MoveOn's email list):\n"
+        args.SLACK_MESSAGE_TEXT += "\n".join([
+            "• %s (%s)" % (
+                account.get('name', '[NO NAME]'),
+                account.get('email', '[NO EMAIL]')
+            )
+            for account in approved_accounts
```
Comment (sjwmoveon) [inline post_report.py]: Nit - not a fan of defining variables after they are used (unless this is idiomatic for Python)

File: post_report.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    if len(declined_accounts):
+        args.SLACK_MESSAGE_TEXT = "The following pending Mobilize accounts "
+        args.SLACK_MESSAGE_TEXT += "should be *declined* "
+        args.SLACK_MESSAGE_TEXT += "(not subscribed to MoveOn's email list):\n"
+        args.SLACK_MESSAGE_TEXT += "\n".join([
+            "• %s (%s)" % (
+                account.get('name', '[NO NAME]'),
+                account.get('email', '[NO EMAIL]')
+            )
+            for account in declined_accounts
```
Comment (sjwmoveon) [inline post_report.py]: See previous.

File: post_report.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+REQUIRED_ARGS = [
+    'DB_HOST', 'DB_PORT', 'DB_USER', 'DB_PASS', 'DB_NAME',
+    'MOBILIZE_API_KEY', 'MOBILIZE_API_SECRET', 'MOBILIZE_API_ROOT',
+    'MOBILIZE_DEFAULT_GROUP_ID', 'SLACK_WEBHOOK', 'SLACK_CHANNEL'
+]
+
+
+def post_report(args) -> list:
```
Comment (sjwmoveon) [inline post_report.py:40]: Can you write a unit test to show we get the output we expect here? [I think this would either require mocking get_pending_accounts somehow or passing in accounts as an argument, but better now while the project is still small...]

File: get_pending_accounts.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                args.MOBILIZE_API_SECRET
+            )
+        )
+        if args.VERBOSE:
+            print('status: %s' % response.status_code)
+        if response.status_code == 200:
+            users = response.json()
+            count += len(users)
+            pending_accounts += [
+                user for user in users
```
Comment (sjwmoveon) [inline get_pending_accounts.py:53]: The pending_accounts assignment is a bit hard to parse; could you split some of this out?  (If this is idiomatic for python please ignore.)  Moving the for loop and if statement to the outside would make it a lot easier for me to follow.

Comment (sjwmoveon) [reply get_pending_accounts.py:53]: Also - the parsing code *could* have its own unit test (with a mock response as the input), but this is simple enough I'm not sure it's 100% needed.

**Review by sjwmoveon — COMMENTED** [review]
Looks pretty good overall; a couple bits were confusing to read and it'd be nice to have at least one test.

File: post_report.py (hunk 1 above)
Comment (sreynen) [reply post_report.py]: I think this is idiomatic for Python, at least as long as we're open to using list comprehensions: https://docs.python.org/3.7/tutorial/datastructures.html#list-comprehensions

It's possible to avoid list comprehensions to avoid variables definitions happening after use, e.g.:
```message_parts = []
for account in approved_accounts:
    message_parts.append("• %s (%s)" % (
          account.get('name', '[NO NAME]'),
          account.get('email', '[NO EMAIL]')
    ))
args.SLACK_MESSAGE_TEXT += "\n".join(message_parts)
```

It's also possible to do less logic inside the list comprehension, so the gap between use and definition is minimal, e.g.:

`args.SLACK_MESSAGE_TEXT += "\n".join([account_description(account) for account in accounts])`

I personally like that last option best, so I'm going to switch to that for now. I'm also open to following another style choice but I think that should be something we decide across all our code and everyone writing that code, as 2 people agreeing on something in a single PR isn't much of a standard.

**Comment by sreynen** [comment]
I added a test, fixed a bug revealed by the new test, and shortened the list comprehensions to make them more readable.

**Review by sjwmoveon — APPROVED** [review]
Looks good.
